contrib/opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -250,11 +173,101 @@ def _transcribe(item, parents, editRate, masterMobs):
+            return timecode
+
+        def _find_mob_chain_and_timecode(source_clip, editRate):
+            """ This is combining several processes in to one for efficieny.
+                 - Firstly, it traverses the SourceClip's Mob's structure using the
+                clip_id to return all the necessary Mobs.
+                 - Secondly, it extracts the necessary Start values for each object, and
+                converts the rate if necessary. This value can come from several different
+                places.
+                 - Lastly, it stores the individual SourceClips so that at the end of the
+                process the correct Length can be derived from the MasterMob"""
+
+            if not isinstance(source_clip, aaf2.components.SourceClip):
+                raise AAFAdapterError("Error: _find_mob_chain() requires a SourceClip component"
+                                      " - cannot parse {}".format(type(source_clip)))
+
+            mob_chain = [source_clip.mob]
+            source_clip_chain = []
+            timecode_chain = []
+            slot_id = source_clip.slot_id
+
+            for mob in mob_chain:
+                if hasattr(mob, 'descriptor') \
+                        and mob.descriptor.name in ['TapeDescriptor', 'ImportDescriptor']:
+                    timecode_chain.append(_find_mobslot_timecode(mob, 1))
+                    # Doing this so mob_chain and source_clip_chain will always have some
+                    # number of values for zip function below
+                    source_clip_chain.append(None)
+                elif mob:
+                    slot = [s for s in mob.slots if s.slot_id == slot_id]
+                    slot = slot[0] if len(slot) > 0 else None
+                    mob_source_clip = _find_source_clip(slot.segment) if slot else None
+                    if mob_source_clip and mob_source_clip.mob:
+                        slot_id = mob_source_clip.slot_id
+                        mob_chain.append(mob_source_clip.mob)
+                        source_clip_chain.append(mob_source_clip)
+                        timecode_chain.append(_find_mobslot_timecode(mob, slot['PhysicalTrackNumber'].value))
+                        timecode_chain.append((mob_source_clip.start, slot.edit_rate.__float__()))
+
+            frame_count = 0
+            for start, rate in timecode_chain:
+                if start == 0:
+                    continue
+                elif rate == editRate:
+                    frame_count += start
+                else:
+                    if rate <= 0:
+                        raise AAFAdapterError("Error: an incompatible rate of {} has been found when "
+                                              "attempting to calculate the source timecode value for "
+                                              "clip {}".format(rate, mob_chain[0].name))
+                    frame_count += int(round(start / (rate / editRate)))
```

Comment:
The only things I have to go off are the values coming out of PyAAF, compared to the values that are in the EDLs I'm exporting from Media Composer. This is currently giving me the correct results for all the examples I have. As for whether it may cause a rounding error somewhere down the track, it's hard to say, but I don't think so. So far, all examples have been pretty comfortably snapping following standard rounding logic.